packages/contracts/foundry_tests/IncentivesController3.t.sol

```diff
@@ -0,0 +1,404 @@
+
+        users.push(makeAddr("eve"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(3 * 365 days);
+
+        users.push(makeAddr("joe"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(4 * 365 days);
+
+        users.push(makeAddr("mike"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(5 * 365 days);
+    }
+
+    function testUpgrade() public {
+        bytes32 oldImpl = vm.load(addressesProvider.getIncentivesController(), IMPLEMENTATION_SLOT);
+
+        _upgradeIncentivesController();
+
+        assertFalse(oldImpl == vm.load(addressesProvider.getIncentivesController(), IMPLEMENTATION_SLOT));
+        assertEq(incentivesController.VE_VMEX(), VE_VMEX);
+        assertEq(incentivesController.DVMEX_REWARD_POOL(), DVMEX_REWARD_POOL);
+        assertEq(incentivesController.DVMEX(), DVMEX);
+    }
+
+    function testQueueNewRewards() public {
+        _upgradeIncentivesController();
+
+        DataTypes.ReserveData memory reserveData = lendingPool.getReserveData(TOKEN_USDC, LP_TRANCHE_ID);
+        _queueNewRewards(reserveData.aTokenAddress, REWARD_AMOUNT);
+
+        IncentivesController.DVmexReward memory reward = incentivesController.getDVmexReward(reserveData.aTokenAddress);
+
+        assertEq(reward.periodFinish, uint32(block.timestamp + 14 days));
+        assertEq(reward.lastUpdateTime, uint32(block.timestamp));
+        assertEq(reward.rewardRate, uint64(REWARD_AMOUNT / 14 days));
+        assertEq(reward.rewardPerTokenStored, 0);
+        assertEq(reward.queuedRewards, 0);
+        assertEq(reward.currentRewards, uint128(REWARD_AMOUNT));
+    }
+
+    function testClaimMinBoost() public {
+        _upgradeIncentivesController();
+
+        DataTypes.ReserveData memory reserveData = lendingPool.getReserveData(TOKEN_USDC, LP_TRANCHE_ID);
+
+        _queueNewRewards(reserveData.aTokenAddress, REWARD_AMOUNT);
+
+        _deposit(USER, TOKEN_USDC, LP_TRANCHE_ID, 1e9);
+
+        vm.warp(block.timestamp + 5 days);
+
+        uint256 earned = incentivesController.earned(reserveData.aTokenAddress, USER);
+        uint256 penaltyBalanceBefore = IERC20(DVMEX).balanceOf(DVMEX_REWARD_POOL);
+
+        vm.prank(USER);
+        incentivesController.claimDVmexReward(reserveData.aTokenAddress, USER);
+
+        IncentivesController.DVmexReward memory rewardInfo =
+            incentivesController.getDVmexReward(reserveData.aTokenAddress);
+        uint256 maxEarned = _standardizeDecimals(
+            IScaledBalanceToken(reserveData.aTokenAddress).scaledBalanceOf(USER), TOKEN_USDC_DECIMALS
+        ) * rewardInfo.rewardPerTokenStored / PRECISION_FACTOR;
+
+        assertEq(IERC20(DVMEX).balanceOf(USER), earned);
```

Comment:
These tests would still pass if earned was zero. Add some validation that the earned amount is somewhat close to expected (or just check its not zero)